From 47437a088d4e7bc9406e0815384c519c9b92b4eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gy=C3=B6rgy=20Ball=C3=B3?= <ballogyor@gmail.com>
Date: Sat, 4 Jul 2015 13:36:25 +0200
Subject: [PATCH] Fix monitoring settings

We have to read settings once, otherwise changes are not detected. Simple reordering fix the problem.
---
 usr/lib/blueberry/blueberry-tray.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/usr/lib/blueberry/blueberry-tray.py b/usr/lib/blueberry/blueberry-tray.py
index 29247ff..e7659e1 100755
--- a/usr/lib/blueberry/blueberry-tray.py
+++ b/usr/lib/blueberry/blueberry-tray.py
@@ -18,14 +18,13 @@ def __init__(self):
 
         self.rfkill = rfkillMagic.Interface(self.update_icon_callback, debug)
         self.settings = blueberrySettings.Settings()
+        self.settings.gsettings.connect("changed::tray-enabled", self.on_settings_changed_cb)
 
         # If we have no adapter, or disabled tray, end early
         if (not self.rfkill.have_adapter) or (not self.settings.get_tray_enabled()):
             self.rfkill.terminate()
             sys.exit(0)
 
-        self.settings.gsettings.connect("changed::tray-enabled", self.on_settings_changed_cb)
-
         self.client = GnomeBluetooth.Client()
         self.model = self.client.get_model()
         self.model.connect('row-changed', self.update_icon_callback)
